src/ann_solo/reader.py

```diff
@@ -259,6 +265,200 @@ def verify_extension(supported_extensions: List[str], filename: str) -> None:
         raise FileNotFoundError(f'File {filename} does not exist')
 
 
+
+
+def read_mzml(source: Union[IO, str], scan_nrs: Sequence[int] = None) \
```

Comment:
**suggestion:** I don't think we need functionality to filter by scan number. This was useful in the GLEAMS code that I shared, but with ANN-SoLo we're always processing all spectra in a file, so we can simplify the code a bit.